src/app/content/utils.ts

```diff
@@ -1,2 +1,32 @@
+import { HTMLElement } from '@openstax/types/lib.dom';
+
 export const stripIdVersion = (id: string): string => id.split('@')[0];
 export const getIdVersion = (id: string): string | undefined => id.split('@')[1];
+
+export const scrollTocSectionIntoView = (sidebar: HTMLElement | undefined, activeSection: HTMLElement | undefined) => {
+  if (!activeSection || !sidebar) {
+    return;
+  }
+
+  // do nothing if the LI is already visible
+  if (activeSection.offsetTop > sidebar.scrollTop
+    && activeSection.offsetTop - sidebar.scrollTop < sidebar.offsetHeight) {
+    return;
+  }
+
+  let search = activeSection.parentElement;
+  let selectedChapter: undefined | HTMLElement;
+  while (search && !selectedChapter && search !== sidebar) {
+    if (search.nodeName === 'LI') {
```

Comment:
i'm not sure what you're getting at, are you concerned with the dependence on the element structure here?

Reply:
Yep, I was concerned about relying on this implicit dependency outside of React but it is less of a concern because of all the tests we have; if the DOM changes, a test will fail and alert that this needs to be fixed

Reply:
not sure what you mean by `this implicit dependency outside of React`. this function was originally in the sidebar react component, but it was a mega hassle to test so i extracted it here